mailpoet/tests/integration/Segments/WPUserWithExtraProps.php

```diff
@@ -4,4 +4,20 @@
 
 class WPUserWithExtraProps extends \WP_User {
   public $orderId;
+
+  /**
+   * add_role contains the 'add_role' hook, which creates side effects,
+   * we do not want. Therefore, we overwrite this method.
```

Comment:
Could you maybe explain it in the comment even a bit more? I'm afraid later on we'll forget what side effects and why we don't want them.

And relatedly:
> Question to me would be if we should rename WPUserWithExtraProps 🤔

I think yes, we can use some more generic name like `WPTestUser` or something like that.

Reply:
Thanks for the review @JanJakes. Addressed in [f616896](https://github.com/mailpoet/mailpoet/pull/4121/commits/f6168968b6fcf126e199e1093acbe2bca89daec6)